### PR TITLE
Users/vinca/pdfix m140

### PR DESCRIPTION
**Pull request by vincentdass**
#8093 #8182

### CODE REVIEW

File: Tasks/AzureRmWebAppDeploymentV4/operations/KuduServiceUtility.ts [hunk 1]
```diff
@@ -78,8 +77,13 @@ export class KuduServiceUtility {
...
         finally {
             try {
+                let deleteFilePath = '%Home%\\site\\VSTS_PostDeployment_' + uniqueID + '\\delete_log_file' + fileExtension;
+                if(taskParams.isLinuxApp) {
+                    deleteFilePath = '/home/site/VSTS_PostDeployment_' + uniqueID + '/delete_log_file' + fileExtension;
+                }
+                
                 await this._appServiceKuduService.uploadFile(vstsPostDeploymentFolderPath, 'delete_log_file' + fileExtension, path.join(__dirname, '..', 'postDeploymentScript', 'deleteLogFile' + fileExtension));
-                await this.runCommand(vstsPostDeploymentFolderPath, 'delete_log_file' + fileExtension, 0, null);
+                await this.runCommand(vstsPostDeploymentFolderPath, deleteFilePath);
```
Comment (SumiranAgg) [inline Tasks/AzureRmWebAppDeploymentV4/operations/KuduServiceUtility.ts]: should we append file extension?

File: Tasks/AzureRmWebAppDeploymentV3/task.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
     "version": {
         "Major": 3,
         "Minor": 4,
-        "Patch": 6
+        "Patch": 9
```
Comment (SumiranAgg) [inline Tasks/AzureRmWebAppDeploymentV3/task.json]: Why are we skipping patch versions 7 and 8?

File: Tasks/AzureRmWebAppDeploymentV4/task.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     "version": {
         "Major": 4,
         "Minor": 3,
-        "Patch": 5
+        "Patch": 8
```
Comment (SumiranAgg) [inline Tasks/AzureRmWebAppDeploymentV4/task.json]: Again is this the patch version we want?

File: Tasks/AzureRmWebAppDeploymentV3/task.json (hunk 2 above)
Comment (vincentdass) [reply Tasks/AzureRmWebAppDeploymentV3/task.json]: cherrypicked from master. so had the same version.

File: Tasks/AzureRmWebAppDeploymentV4/operations/KuduServiceUtility.ts (hunk 1 above)
Comment (vincentdass) [reply Tasks/AzureRmWebAppDeploymentV4/operations/KuduServiceUtility.ts]: deleteFilePath variable has the extension included.

File: Tasks/AzureRmWebAppDeploymentV4/task.json (hunk 3 above)
Comment (vincentdass) [reply Tasks/AzureRmWebAppDeploymentV4/task.json]: we can skip this.

Comment (vincentdass) [reply Tasks/AzureRmWebAppDeploymentV4/task.json]: thanks, updated.

File: Tasks/AzureRmWebAppDeploymentV3/operations/TaskParameters.ts [hunk 4]
```diff
@@ -52,8 +53,25 @@ export class TaskParametersUtility {
...
         return taskParameters;
     }
+
+    private static UpdateLinuxAppTypeScriptParameters(taskParameters: TaskParameters) {
+        let retryTimeoutValue = tl.getVariable('appservicedeploy.retrytimeout');
+        let timeoutAppSettings = retryTimeoutValue ? Number(retryTimeoutValue) * 60 : 1800;
+
+        tl.debug(`setting app setting SCM_COMMAND_IDLE_TIMEOUT to ${timeoutAppSettings}`);
+        if(taskParameters.AppSettings) {
+            taskParameters.AppSettings = taskParameters.AppSettings + ` -SCM_COMMAND_IDLE_TIMEOUT ${timeoutAppSettings}`;
```
[the remark this answers is not in the document]
Comment (vincentdass) [reply Tasks/AzureRmWebAppDeploymentV3/operations/TaskParameters.ts]: done.

File: Tasks/Common/azure-arm-rest/azure-arm-app-service-kudu.ts [hunk 5]
```diff
@@ -395,7 +395,8 @@ export class Kudu {
 
         try {
             tl.debug('Executing Script on Kudu. Command: ' + command);
-            var response = await this._client.beginRequest(httpRequest);
+            let webRequestOptions: webClient.WebRequestOptions = {retriableErrorCodes: [], retriableStatusCodes: [], retryCount: 0, retryIntervalInSeconds: 5};
```
[the remark this answers is not in the document]
Comment (vincentdass) [reply Tasks/Common/azure-arm-rest/azure-arm-app-service-kudu.ts]: done

**Pull request merged**